### PR TITLE
test(sandbox): harden TempPathGuard cleanup for symlink paths

### LINKED ISSUE #194: Make sandbox fixture TempPathGuard cleanup symlink-safe
## Context
Gemini review on PR #188 suggested strengthening temporary fixture cleanup behavior.

## Problem
`TempPathGuard` cleanup currently relies on file type checks that may follow symlinks in edge cases. Test harness cleanup should be symlink-safe and deterministic.

## Scope
- Update `TempPathGuard` cleanup in sandbox security fixtures to use symlink-safe metadata checks.
- Ensure directory/file removal behavior is explicit and resilient to symlink edge cases.
- Add/adjust tests if needed to lock behavior.

## Done When
- Fixture cleanup is symlink-safe and deterministic.
- No regression in fixture lifecycle behavior.
- `cargo fmt`, `cargo clippy`, and `cargo test -p clawcrate-sandbox --test security_fixtures` pass.

### PR DESCRIPTION
**Pull request by manuelpenazuniga**
## Summary
- harden `TempPathGuard` cleanup to use `symlink_metadata` so fixture teardown does not follow symlinks implicitly
- keep deterministic cleanup behavior for files, directories, symlinks, and not-found paths
- add Linux fixture test proving guard removal of symlink path does not delete the underlying target directory

## Validation
- `cargo fmt --all`
- `cargo clippy -p clawcrate-sandbox --all-targets -- -D warnings`
- `cargo test -p clawcrate-sandbox`

Closes #194

### CODE REVIEW

File: crates/clawcrate-sandbox/tests/security_fixtures.rs [hunk 1]
```diff
@@ -138,10 +140,20 @@ impl TempPathGuard {
...
+                } else {
+                    let _ = fs::remove_file(&self.path);
+                }
+            }
+            Err(error) => {
+                if error.kind() != io::ErrorKind::NotFound {
+                    let _ = error;
+                }
+            }
         }
```
[the remark this answers is not in the document]
Comment (manuelpenazuniga) [reply crates/clawcrate-sandbox/tests/security_fixtures.rs:157]: ok
